packages/vm/src/imports.rs

```diff
@@ -98,8 +98,30 @@ pub fn do_db_write<A: BackendApi + 'static, S: Storage + 'static, Q: Querier + '
         return Err(VmError::write_access_denied());
     }
 
-    let key = read_region(&data.memory(&store), key_ptr, MAX_LENGTH_DB_KEY)?;
-    let value = read_region(&data.memory(&store), value_ptr, MAX_LENGTH_DB_VALUE)?;
+    /// Converts a region length error to a different variant for better understandability
+    fn convert_error(e: VmError, kind: &'static str) -> VmError {
+        if let VmError::CommunicationErr {
+            source: CommunicationError::RegionLengthTooBig { length, max_length },
+            #[cfg(feature = "backtraces")]
+            backtrace,
+        } = e
+        {
+            VmError::WriteTooBig {
+                kind,
+                length,
+                max_length,
+                #[cfg(feature = "backtraces")]
+                backtrace,
+            }
+        } else {
+            e
+        }
+    }
+
+    let key = read_region(&data.memory(&store), key_ptr, MAX_LENGTH_DB_KEY)
+        .map_err(|e| convert_error(e, "Key"))?;
+    let value = read_region(&data.memory(&store), value_ptr, MAX_LENGTH_DB_VALUE)
+        .map_err(|e| convert_error(e, "Value"))?;
```

Comment:
Can't you just map `RegionLengthTooBig` to `VmError::GenericError` in those two calls, knowing that only the error text is preserved anyways for import implementations? This would avoid creating a new structured error type which is never processed in a structured way.

Reply:
Good point, yes we can do that.